substrate/primitives/application-crypto/test/src/ecdsa.rs

```diff
@@ -44,3 +46,12 @@ fn ecdsa_works_in_runtime() {
 	assert!(supported_keys.contains(&public.to_raw_vec()));
 	assert!(AppPair::verify(&signature, "ecdsa", &public));
 }
+
+#[test]
+fn ecdsa_client_generated_pop_verified_by_runtime_public_crypto() {
```

Comment:
Here an example of what I mean we need two tests per crypto one generate by client verify by runtimeapp and vice versa.